File: spark-app/src/main/java/com/kong/spark/DataFrameOps.java
package com.kong;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;

/**
 * Created by kong on 2016/3/14 0014.
 */
public class DataFrameOps {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("sparkSQL");
        JavaSparkContext sc = new JavaSparkContext(conf);
        SQLContext sqlContext = new SQLContext(sc);
        //创建DataFrame，可以简单的认为DataFrame是一张表
        DataFrame df = sqlContext.read().json("");
        //select * from table
        df.show();
        //desc table
        df.printSchema();
        //select name from table
        df.select("name").show();
        //select name,age+1 from table
        df.select(df.col("name"), df.col("age").plus(10)).show();
        //select * from table where age > 10
        df.filter(df.col("age").gt(10)).show();
        //select count(1) from table group by age
        df.groupBy(df.col("age")).count().show();
    }
}

File: spark-app/src/main/scala/com/kong/core/sql/DataFrameOps.scala
package com.kong.core.sql

import org.apache.spark.sql.SQLContext
import org.apache.spark.{SparkContext, SparkConf}

/**
 * Created by kong on 2016/3/14 0014.
 */
object DataFrameOps {
  def main(args: Array[String]) {
    val conf = new SparkConf().setAppName("Scala DataFrame")
    val sc = new SparkContext(conf)
    val sql = new SQLContext(sc)
    val df = sql.read.json("hdfs://Master:9000")
    df.show
    df.printSchema
    df.select("name").show
    df.select(df("name"), df("age") + 10).show
    df.filter(df("age") > 10).show
    df.groupBy("age").count.show

    sc.stop()
  }
}
